src/probability_theory/density.lean

```diff
@@ -0,0 +1,339 @@
+    exactI ⟨have_lebesgue_decomposition_of_has_pdf, map_absolutely_continuous⟩ },
+  { rintros ⟨h_decomp, h⟩,
+    haveI := h_decomp,
+    refine ⟨⟨hX, (measure.map X ℙ).rn_deriv μ, measurable_rn_deriv _ _, _⟩⟩,
+    rwa with_density_rn_deriv_eq }
+end
+
+section
+
+variables {F : Type*} [normed_group F] [measurable_space F] [second_countable_topology F]
+  [normed_space ℝ F] [complete_space F] [borel_space F] {ν : measure F}
+
+/-- A random variable that `has_pdf` transformed under a `quasi_measure_preserving`
+map also `has_pdf` if `(map g (map X ℙ)).have_lebesgue_decomposition μ`.
+
+`quasi_measure_preserving_has_pdf'` is more useful in the case we are working with a
+probability measure and a real-valued random variable. -/
+lemma quasi_measure_preserving_has_pdf {X : α → E} [has_pdf X ℙ μ]
+  {g : E → F} (hg : quasi_measure_preserving g μ ν)
+  (hmap : (map g (map X ℙ)).have_lebesgue_decomposition ν) :
+  has_pdf (g ∘ X) ℙ ν :=
+begin
+  rw [has_pdf_iff, ← map_map hg.measurable (has_pdf.measurable X ℙ μ)],
+  refine ⟨hg.measurable.comp (has_pdf.measurable X ℙ μ), hmap, _⟩,
+  rw [map_eq_with_density_pdf X ℙ μ],
+  refine absolutely_continuous.mk (λ s hsm hs, _),
+  rw [map_apply hg.measurable hsm, with_density_apply _ (hg.measurable hsm)],
+  have := hg.absolutely_continuous hs,
+  rw map_apply hg.measurable hsm at this,
+  exact set_lintegral_measure_zero _ _ this,
+end
+
+lemma quasi_measure_preserving_has_pdf' [is_finite_measure ℙ] [sigma_finite ν]
+  {X : α → E} [has_pdf X ℙ μ] {g : E → F} (hg : quasi_measure_preserving g μ ν) :
+  has_pdf (g ∘ X) ℙ ν :=
+begin
+  haveI : is_finite_measure (map g (map X ℙ)) :=
+    @is_finite_measure_map _ _ _ _ (map X ℙ)
+      (is_finite_measure_map ℙ (has_pdf.measurable X ℙ μ)) _ hg.measurable,
+  exact quasi_measure_preserving_has_pdf hg infer_instance,
+end
+
+end
+
+section real
+
+variables [is_finite_measure ℙ] {X : α → ℝ}
+
+/-- A real-valued random variable `X` `has_pdf X ℙ λ` (where `λ` is the Lebesgue measure) if and
+only if the push-forward measure of `ℙ` along `X` is absolutely continuous with respect to `λ`. -/
+lemma real.has_pdf_iff_of_measurable (hX : measurable X) : has_pdf X ℙ ↔ map X ℙ ≪ volume :=
+begin
+  haveI : is_finite_measure (map X ℙ) := is_finite_measure_map ℙ hX,
+  rw [has_pdf_iff_of_measurable hX, and_iff_right_iff_imp],
+  exact λ h, infer_instance,
+end
+
+lemma real.has_pdf_iff : has_pdf X ℙ ↔ measurable X ∧ map X ℙ ≪ volume :=
+begin
+  by_cases hX : measurable X,
+  { rw [real.has_pdf_iff_of_measurable hX, iff_and_self],
+    exact λ h, hX,
+    apply_instance },
+  { split,
+    { exact λ h, false.elim (hX h.pdf'.1) },
+    { exact λ h, false.elim (hX h.1) } }
+end
+
+/-- If `X` is a real-valued random variable that has pdf `f`, then the expectation of `X` equals
+`∫ x, x * f x ∂λ` where `λ` is the Lebesgue measure. -/
+lemma integral_mul_eq_integral [has_pdf X ℙ] :
+  ∫ x, x * (pdf X ℙ volume x).to_real = ∫ x, X x ∂ℙ :=
+integral_mul_eq_integral' measurable_id
```

Comment:
Do you think we could come up with more enlightening names here? It's not particularly clear to me that removing a prime should corresponding to specialising to `g = id`!

Not essential.

Reply:
I've changed it to `integral_fun_mul_eq_integral`